lib/Agrammon/Web/Routes.pm6

```diff
@@ -145,6 +145,24 @@ sub api-routes (Str $schema, $ws) {
                 }
             }
         }
+        operation 'changePassword', -> LoggedIn $user {
+            request-body -> (:oldPassword($old-password)!, :newPassword($new-password)!) {
+                $ws.change-password($user, $old-password, $new-password);
+                CATCH {
+                    note "$_";
+                    when X::Agrammon::DB::User::PasswordsIdentical {
+                        conflict 'application/json', %(
+                            error => .message
+                        );
+                    }
+                    when X::Agrammon::DB::User::InvalidPassword {
+                        response.status = 422;
+                        response.append-header('Content-type', 'application/json');
+                        response.set-body(to-json { :error(.message) });
```

Comment:
You can actually just write `content 'application/json', %( error => .message )` (and note you can't use curly syntax here because you reference `$_` which makes it a closure). `content` will only set a `200` status if no status is already set. So in fact `conflict` is implemented by setting status and then calling `content`.

Reply:
Ah, I was hoping for an easier way :-)

Reply:
Fixed